### PR TITLE
Stabilize random number reproducibility in doctests

### PR DESCRIPTION
**Pull request by nathanrboyer**
`MersenneTwister(1234)` from Random.jl yields different results now than those shown in the docstrings. I updated the docstrings of `shuffle` and `shuffle!` so that their output matches the new results. I also had to modify the `docs` Project.toml to be able to run the doctests correctly as suggested by [Documenter.jl](https://documenter.juliadocs.org/dev/man/guide/#Package-Guide).

> Note that for packages, you also likely need to have your package that you are documenting as a ["dev dependency"](https://pkgdocs.julialang.org/v1/managing-packages/#developing) of the docs/ environment.

This should fix the currently failing doctests in #3360.

### CODE REVIEW

**Comment by nathanrboyer** [comment]
This is what I ran from within the DataFrames directory which updated the docs Project.toml file and enabled me to run `julia --project make.jl`:

```julia
(DataFrames) pkg> activate docs

(docs) pkg> dev .
```

I'm not sure if that relationship is fully captured by the code in the Project.toml file.

**Comment by nathanrboyer** [comment]
Now the behavior flipped so this branch is failing and #3360 passed???

**Comment by nathanrboyer** [comment]
Hey @bkamins, not trying to bug you, but I wanted to ping your inbox since this PR and #3360 have been ready for a little while now.

**Comment by nathanrboyer** [comment]
According to a [recent discourse post](https://discourse.julialang.org/t/julia-1-11-1-gives-different-results-from-1-10-5/123433/53), seeded random numbers can change between Julia versions. The [StableRNGs.jl](https://github.com/JuliaRandom/StableRNGs.jl) package was created for reproducible testing as used here, so I attempted to implement it. 

I'm not sure if all of the calls to `Random.seed!` should also change to use `StableRNG`, but the doctests are passing for me as is.

**Comment by bkamins** [comment]
@nathanrboyer thank you for working on it. I think we also need an entry to `[compat'` section for StableRNGs to fix its version. Other than that it looks good.

Then let us merge this PR and merge #360 with `main` after this merge it so that I can review #360 in a clean way (it is a big PR so it is better to avoid risks of later merge conflicts).

**Comment by nathanrboyer** [comment]
I added a compat entry to the docs project.toml. I don't know of a way to do the same for test dependencies added as "extras"?

File: Project.toml [hunk 1]
```diff
@@ -60,8 +60,21 @@ Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
 OffsetArrays = "6fe1bfb0-de20-5000-8ca7-80f57d26f881"
 ShiftedArrays = "1277b4bf-5013-50f5-be3d-901d8477a67a"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
+StableRNGs = "860ef19b-820b-49d6-a774-d7a799459cd3"
```
Comment (bkamins) [inline Project.toml:63]: We need to also add these dependencies here as all extras entreis should have info in this file (we use them in tests not only in documentation building). Thank you!

**Review by bkamins — APPROVED** [review]

**Pull request merged**